lib/jekyll-github-metadata/repository.rb

```diff
@@ -130,56 +130,26 @@ def default_user_domain
         end
       end
 
-      def user_page_domains
-        domains = [default_user_domain]
-        domains.push "#{owner}.github.com".downcase unless Pages.enterprise?
-        domains
+      def cname
+        return nil unless Pages.custom_domains_enabled?
+        repo_pages_info["cname"]
       end
 
-      def user_domain
-        domain = default_user_domain
-        user_page_domains.each do |user_repo|
-          candidate_nwo = "#{owner}/#{user_repo}"
-          next unless Value.new(proc { |client| client.repository? candidate_nwo }).render
-          domain = self.class.new(candidate_nwo).domain
-        end
-        domain
-      end
-
-      def pages_url
-        if !Pages.custom_domains_enabled?
-          path = user_page? ? owner : nwo
-          if Pages.subdomain_isolation?
-            URI.join("#{Pages.scheme}://#{Pages.pages_hostname}/", path).to_s
-          else
-            URI.join("#{Pages.github_url}/pages/", path).to_s
-          end
-        elsif cname || primary?
-          "#{url_scheme}://#{domain}"
-        else
-          URI.join("#{url_scheme}://#{domain}", name).to_s
-        end
+      def html_url
+        @html_url ||= (repo_pages_info["html_url"] || repo_compat.pages_url)
```

Comment:
@benbalter as well as this. this is where it grabs the `html_url` from the pages endpoint if it exists, otherwise uses the old `pages_url` like before.